src/Compilers/CSharp/Test/Semantic/Semantics/NullableReferenceTypesTests.cs

```diff
@@ -28476,6 +28476,263 @@ void M2()
+                Diagnostic(ErrorCode.WRN_DisallowNullAttributeForbidsMaybeNullAssignment, "(B?)null").WithLocation(18, 12));
+        }
+
+        [Fact]
+        [WorkItem(39802, "https://github.com/dotnet/roslyn/issues/39802")]
+        public void UserDefinedConversion_UnconditionalParamAttributes_Lifted()
+        {
+            var source = @"
+using System.Diagnostics.CodeAnalysis;
+
+public struct A { }
+
+public struct B
+{
+    public static implicit operator A([DisallowNull] B b)
+    {
+        return new A();
+    }
+
+    void M0(A? a) { }
+    void M1([DisallowNull] A? a) { }
+
+    void M2()
+    {
+        M0((B?)new B());
+        M0((B?)null);
+
+        M1((B?)new B());
+        M1((B?)null); // 1
+    }
+}
+
+public struct C
+{
+    [return: NotNull]
+    public static implicit operator A?([DisallowNull] C c)
+    {
+        return new A();
+    }
+
+    void M0(A? a) { }
+    void M1([DisallowNull] A? a) { }
+
+    void M2()
+    {
+        M0((C?)new C());
+        M0((C?)null);
+
+        M1((C?)new C());
+        M1((C?)null); // 2
+    }
+}
+
+public struct D
+{
+    [return: NotNullIfNotNull(""d"")]
+    public static implicit operator A?([DisallowNull] D d)
+    {
+        return new A();
+    }
+
+    void M0(A? a) { }
+    void M1([DisallowNull] A? a) { }
+
+    void M2()
+    {
+        M0((D?)new D());
+        M0((D?)null);
+
+        M1((D?)new D());
+        M1((D?)null); // 3
+    }
+}
+";
+            var comp = CreateNullableCompilation(new[] { AllowNullAttributeDefinition, DisallowNullAttributeDefinition, NotNullAttributeDefinition, NotNullIfNotNullAttributeDefinition, source });
+            comp.VerifyDiagnostics(
+                // (22,12): warning CS8607: A possible null value may not be used for a type marked with [NotNull] or [DisallowNull]
+                //         M1((B?)null); // 1
+                Diagnostic(ErrorCode.WRN_DisallowNullAttributeForbidsMaybeNullAssignment, "(B?)null").WithLocation(22, 12),
+                // (43,12): warning CS8607: A possible null value may not be used for a type marked with [NotNull] or [DisallowNull]
+                //         M1((C?)null); // 2
+                Diagnostic(ErrorCode.WRN_DisallowNullAttributeForbidsMaybeNullAssignment, "(C?)null").WithLocation(43, 12),
+                // (64,12): warning CS8607: A possible null value may not be used for a type marked with [NotNull] or [DisallowNull]
+                //         M1((D?)null); // 3
+                Diagnostic(ErrorCode.WRN_DisallowNullAttributeForbidsMaybeNullAssignment, "(D?)null").WithLocation(64, 12));
+        }
+
+        [Fact]
+        [WorkItem(39802, "https://github.com/dotnet/roslyn/issues/39802")]
+        public void UserDefinedConversion_UnconditionalReturnAttributes()
+        {
+            var source = @"
+using System.Diagnostics.CodeAnalysis;
+
+public struct A { }
+public struct B
+{
+    [return: NotNull]
+    public static implicit operator A?(B? b)
```

Comment:
Consider testing `[return: NotNull] public static implicit operator A(B b) => new A();` with conversion from `B?`, to ensure we treat the result as `A?`.